### PR TITLE
FOSSA scan enabled

### PR DESCRIPTION
**Pull request by idvoretskyi**
Ref.: https://github.com/cncf/foundation/issues/109

Signed-off-by: Ihor Dvoretskyi <ihor@linux.com>

### CODE REVIEW

File: .github/workflows/fossa.yml
Comment (milosgajdos) [inline .github/workflows/fossa.yml:18]: Should this be added to GH repo secrets rather than being exposed here?

Comment (idvoretskyi) [reply .github/workflows/fossa.yml:18]: @milosgajdos nope, unfortunately, it won't work as a secret if triggered from the repo fork (https://docs.github.com/en/actions/reference/encrypted-secrets#using-encrypted-secrets-in-a-workflow).

At the same time, this is a read-only API key, which can be exposed safely - https://docs.fossa.com/docs/api-reference#push-only-api-token, so let's keep it as is.

Comment (milosgajdos) [reply .github/workflows/fossa.yml:18]: This is a bit strange. I can see how someone can abuse this easily by pushing all kinds of nonsense using this API key. Does this also mean that forks will be able to override the upstream?

Comment (idvoretskyi) [reply .github/workflows/fossa.yml:18]: @milosgajdos I agree that it could be done differently, but this is how does it work. So it's a common approach across the whole board of CNCF projects (they are all referenced in this issue - https://github.com/cncf/foundation/issues/109).

You can find more details in the thread under this PR https://github.com/kedacore/keda/pull/860, where we've noticed these challenges for the first time.

Comment (wy65701436) [reply .github/workflows/fossa.yml:18]: From the PR [kedacore/keda#860](https://github.com/kedacore/keda/pull/860), it's using the secret as the API key.

Comment (idvoretskyi) [reply .github/workflows/fossa.yml:18]: @wy65701436 correct, we've tried, but it didn't work. We've reverted it here - https://github.com/kedacore/keda/pull/937

You can find the actual working action config file here - https://github.com/kedacore/keda/blob/main/.github/workflows/fossa.yml, and a list of the recently triggered actions here - https://github.com/kedacore/keda/actions/workflows/fossa.yml, for example.

Comment (idvoretskyi) [reply .github/workflows/fossa.yml:18]: @milosgajdos anything else that blocks this PR from being merged?

Comment (milosgajdos) [reply .github/workflows/fossa.yml:18]: From my side, this feels dodgy and I'd like someone to talk to FOSSA developers so they sort out this security circus with GitHub folks.

However, CNCF now own this repo so if they're happy with the current state of things we can merge this in, indeed. Before we do that, can you please remove some empty chars from the PR so the CI completes successfully? You should be able to track those empty chars by running: `git show --check c8e7fbecc6cc105aaa242b1515945f6c746fc21c`

Comment (idvoretskyi) [reply .github/workflows/fossa.yml:18]: > From my side, this feels dodgy and I'd like someone to talk to FOSSA developers so they sort out this security circus with GitHub folks.

I have already discussed this with FOSSA a while ago, and yes, this is their suggested solution.

> However, CNCF now own this repo so if they're happy with the current state of things we can merge this in, indeed. 

To clarify - I am with CNCF. I agree that this is not the most perfect solution that could be, but this is a common practice across other projects; suggested by the FOSSA team themselves.

Comment (milosgajdos) [reply .github/workflows/fossa.yml:18]: Sure, but I can't help but still express my dislike of it. Going forward it's probably something worth bringing up with @chrispat so GH come up with some way of addressing this.

**Review by milosgajdos — APPROVED** [review]
Ignore the CI. The failed run is related to the empty space char present in one of the earlier commits and it's making CI sad. I think if you rebased the CI run would pass, but we can do that during merge.

File: .github/workflows/fossa.yml
Comment (chrispat) [reply .github/workflows/fossa.yml:18]: If you are using a secret in a workflow that is run on pull requests from forks it would easily be extracted so having it as a secret value vs plain text isn't protecting anything.

Comment (milosgajdos) [reply .github/workflows/fossa.yml:18]: Yes, I understand this, but I'm struggling to accept it as an "Ok situation because XYZ". I haven't given this much thought, tbh, but surely it's in GH's [and OSS developers] interest to come up with some solution to this workflow?

Comment (chrispat) [reply .github/workflows/fossa.yml:18]: It would be great if something were possible but I just don't know how to keep any sort of value secret in an environment that lest you run arbitrary code from unknown sources.  In the case of CI and open source contribution that is a fundamental need.

It is possible that FOSSA could enable a system like you have with CodeCov where they use some other information in the environment along with a call back to the Actions API rather than an API key.

**Comment by idvoretskyi** [comment]
@milosgajdos any other blockers for now?

**Comment by milosgajdos** [comment]
> @milosgajdos any other blockers for now?

I gave it my tick a while back. Unfortunately, I can't merge it unless there are at least 2 approvals from maintainers -- otherwise I would have already merged it in.
 
CC: @chrispat @wy65701436

**Comment by milosgajdos** [comment]
Ok, one last bit @idvoretskyi . Can you please squash the commits? I believe that will also get rid of the build failure as there is some empty space char in your first commit (See `git show --check c8e7fbecc6cc105aaa242b1515945f6c746fc21c`) Then we're good to go.

**Comment by idvoretskyi** [comment]
@milosgajdos here we go :)

**Pull request merged**
